lib/parliament/request/open_search_request.rb

```diff
@@ -71,32 +75,41 @@ def get_description(url)
           end
 
           request = Parliament::Request::BaseRequest.new(base_url: url,
-                                                         headers: {
-                                                             'Accept' => 'application/opensearchdescription+xml',
-                                                             'Ocp-Apim-Subscription-Key' => ENV['OPENSEARCH_AUTH_TOKEN']
+                                                         headers:  {
+                                                          'Accept' => 'application/opensearchdescription+xml',
+                                                          'Ocp-Apim-Subscription-Key' => ENV['OPENSEARCH_AUTH_TOKEN']
                                                          })
           xml_response = request.get
 
           begin
             xml_root = REXML::Document.new(xml_response.response.body).root
-            template = xml_root.elements['Url'].attributes['template']
-
-            raise Parliament::OpenSearch::DescriptionError.new(url), "The document found does not contain a require node. Attempted to get a 'Url' element with the attribute 'template'. Please check the description document at '#{url}' and try again." if template.nil?
+            templates = { url: [] }
+            xml_root.elements.each('Url') do |url|
+              type = url.attributes['type']
+              template = url.attributes['template']
+              templates[:url] << { type: type, template: template }
+            end
+
+            raise Parliament::OpenSearch::DescriptionError.new(url), "The document found does not contain the required node. Attempted to get a 'Url' element with the attribute 'template'. Please check the description document at '#{url}' and try again." if templates[:url].any? { |url| url[:template].nil? }
           rescue NoMethodError
             raise Parliament::OpenSearch::DescriptionError.new(url), "The document found does not appear to be XML. Please check the description document at '#{url}' and try again."
           end
 
-          template
+          templates
         end
       end
 
       def query_url
         @query_url
       end
 
-      def setup_query_url(search_params)
+      def setup_query_url(search_params, type = nil)
         search_terms = search_params[:query]
-        query_url = @base_url.dup
+        type ||= 'application/atom+xml'
```

Comment:
should this not just be `type = 'application/atom+xml'` in the parameters above?